msmtools/analysis/sparse/stationary_vector.py

```diff
@@ -118,10 +119,28 @@ def stationary_distribution_from_eigenvector(T, ncv=None):
         Vector of stationary probabilities.
 
     """
-    vals, vecs = scipy.sparse.linalg.eigs(T.transpose(), k=1, which='LR', ncv=ncv)
-    nu = vecs[:, 0].real
-    mu = nu / np.sum(nu)
-    return mu
+
+    # get the top two eigenvalues and vecs so we can check for irreducibility
+    vals, vecs = scipy.sparse.linalg.eigs(T.transpose(), k=2, which='LR', ncv=ncv)
+
+    # check for irreducibility
+    if np.allclose(vals, 1, rtol=EPS, atol=EPS):
```

Comment:
A tolerance of EPS might be too strict, since I often had degenerated ones of reducible matrices that weren't exactly one up to machine precision... How small would the tolerance need to be in your opinion to avoid false raises?

Reply:
I think this needs to be seen through experience - I would start with 1e2*EPS, merge this and adapt as needed.

Reply:
Ok, let's do it so! :)